### ISSUE TITLE
Feature: Allow linting on stdin

### ISSUE THREAD

**Issue by gdavison** [terraform-linters/tflint#858]
In some CI validation usecases, such as blocks in documentation, it could be useful to pipe a configuration to `tflint` on `stdin`.

For example:
```sh
terrafmt blocks --zero-terminated my_docs.md | xargs -0 -n1 tflint --format=json
```

(Null-terminated blocks in `terrafmt` are suggested in https://github.com/katbyte/terrafmt/issues/24)

**Comment by bendrucker** [comment]
Interesting. The provided directory gets two passes: Terraform loads the configuration, and tflint separately scans for annotations. The raw HCL bodies are also re-parsed from the source cache so rules can do low level inspection if desired.

https://github.com/terraform-linters/tflint/blob/master/cmd/inspect.go#L95

I think the most straightforward change here is to load stdin and put it in an in-memory `afero.Fs` as `stdin.tf`. That should minimize the amount of code changing.

**Comment by wata727** [comment]
I'm thinking about this now, but given that TFLint is moving towards a directory-oriented interface, I'm wondering if we should implement this feature.

**Comment by bendrucker** [comment]
In the original example of terrafmt, it seems like there'd need be substantial value in a more advanced wrapper than just gluing together with xargs, especially in GitHub Actions.

If TFLint supported stdin, it would have to use a generated filename like `input.tf` and would report ranges relative to that filename and its lines.

This is not especially helpful for users, who would instead want a GitHub error annotation that is reported on the actual source, e.g., in a Terraform code block in a Markdown file.
